os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -248,3 +251,32 @@ void binary_manager_remove_binlist(FAR struct tcb_s *tcb)
 		if (next) next->bin_blink = prev;
 	}
 }
+
+/****************************************************************************
+ * Name: binary_manager_clear_bindata
+ *
+ * Description:
+ *	 This function clears data of binary with bin_idx in a binary table.
+ *
+ ****************************************************************************/
+void binary_manager_clear_bindata(int bin_idx)
+{
+	struct tcb_s* tcb;
+
+	if (bin_idx < 0 || bin_idx > USER_BIN_COUNT) {
+		return;
+	}
+
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	tcb = sched_gettcb(BIN_ID(bin_idx));
+	if (!((struct task_tcb_s *)tcb)->bininfo->reload) {
```

Comment:
What does this conditional mean?

Reply:
A 'reload' variable in bininfo is used as reloading flag if CONFIG_OPTIMIZE_APP_RELOAD_TIME is enabled only.